### ISSUE TITLE
Perspective tool would be great.

### ISSUE THREAD

**Issue by prandygame** [opentoonz/opentoonz#4755]
### Is your feature request related to a problem? Please describe

I still don't understand how this wonderful program doesn't have a perspective tool.

### Describe the solution you'd like

It would be nice if you could make it easier to create backgrounds and geometric objects by putting some vanishing points, make the thickness of the pen increase as you move away from the vanishing point, and even when you have to draw light poles or something that repeats itself It is enough to draw one so that several more are drawn at the same time respecting the vanishing point, you draw a window and two more are drawn next to it at the same time or whatever you want, it would be something interesting to have for certain moments, thanks for reading my suggestion.

**Comment by RodneyBaker** [comment]
@prandygame I'll guess you don't know the history of perspective tool development for Opentoonz but... there is one.  :)

I suppose the bottom line at present time is that no one has added the tool yet although there have been several approaches developed.

You may wish to try out the Tahoma2D (T2D) fork of Opentoonz as it has a perspective tool.
It had a slightly different one some time back as well (but that can only be found in those early releases).

The Morevna Edition fork aslo should have a perspective tool but it was found to be a little problematic in that it caused some slowdowns.  This unfortunately stopped it's trajectory for being added to Opentoonz.

My own take on the current perspective tools (those that have almost made it into Opentoonz) is that none quite deliver as well as they could. Tahoma2D's perspective feature perhaps comes closest.  I liked the simplicity of the one it had prior to that also.  I have no memory of ever attempting to use the tool in the Morevna Edition so cannot offer much related to that effort.  A lot of work went into that tool and it's sad that it couldn't move forward.

Perhaps you can try out the perspective tool in Tahoma2D and provide feedback on how well it meets your expectations?

In other news...
The Tahoma2D fork has recently gained a symmetry tool as well which is quite nice a versatile.
I've posted a number of drawings I did with that tool in the development channel of the T2D discord.

**Comment by prandygame** [comment]
> 

Thanks for answering, well yes, those forks appeared to me when I was researching, but my projects are unfortunately made in the penultimate version of Opentoonz, the tool does not have to be so complex, at least meanwhile a point or two points of leakage without any another feature mentioned, I'm glad to know that this topic has already been discussed, but I think it's very important that Opentoonz has something that makes it easy to create backgrounds whatever it is, at least I would like to know how to make perfectly horizontal, vertical lines and diagonals?

**Comment by prandygame** [comment]
> @prandygame I'll guess you don't know the history of perspective tool development for Opentoonz but... there is one. :)
> 
> I suppose the bottom line at present time is that no one has added the tool yet although there have been several approaches developed.
> 
> You may wish to try out the Tahoma2D (T2D) fork of Opentoonz as it has a perspective tool. It had a slightly different one some time back as well (but that can only be found in those early releases).
> 
> The Morevna Edition fork aslo should have a perspective tool but it was found to be a little problematic in that it caused some slowdowns. This unfortunately stopped it's trajectory for being added to Opentoonz.
> 
> My own take on the current perspective tools (those that have almost made it into Opentoonz) is that none quite deliver as well as they could. Tahoma2D's perspective feature perhaps comes closest. I liked the simplicity of the one it had prior to that also. I have no memory of ever attempting to use the tool in the Morevna Edition so cannot offer much related to that effort. A lot of work went into that tool and it's sad that it couldn't move forward.
> 
> Perhaps you can try out the perspective tool in Tahoma2D and provide feedback on how well it meets your expectations?
> 
> In other news... The Tahoma2D fork has recently gained a symmetry tool as well which is quite nice a versatile. I've posted a number of drawings I did with that tool in the development channel of the T2D discord.

@RodneyBaker  Thanks for answering, well yes, those forks appeared to me when I was researching, but my projects are unfortunately made in the penultimate version of Opentoonz, the tool does not have to be so complex, at least meanwhile a point or two points of leakage without any another feature mentioned, I'm glad to know that this topic has already been discussed, but I think it's very important that Opentoonz has something that makes it easy to create backgrounds whatever it is, at least I would like to know how to make perfectly horizontal, vertical lines and diagonals?

**Comment by RodneyBaker** [comment]
Regarding the perfectly horizontal, vertical and diagonal lines...

If using one of the latest nightly releases we can now draw straight lines with the aid of helper key.
For more information on that please see:  PR: #4680

**Comment by prandygame** [comment]
@RodneyBaker Thanks!

**Comment by RodneyBaker** [comment]
While perspective tools definitely need to be investigated for use with Opentoonz...

Given that the current (as of 3 days ago) nightly release of the Tahoma2D fork of Opentoonz has this feature will close this request with the intention of building a porting list.  Said porting list to identify ideal features to port from Tahoma2D (and other forks of Opentoonz) with a goal to keep the code between forks as similar as possible.  In cases where this isn't possible we can then document the reasons why a different approach was desired and (where reasonable) point out differences in features and code.

I personally recommend Opentoonz users also maintain a copy of Tahoma2D if for no other reason than it provides an easy means to access third party applications not included with Opentoonz such as FFMPEG, Rhubarb (for lipsync).

Bottom line:  The requested feature is available for immedaite use and the source code is also available for porting should interested developers choose to do so.

Link to Tahoma2D nightly releases:  https://github.com/tahoma2d/tahoma2d_nightlies/releases

**Comment by MYToonz** [comment]
I'm a supporter of adding perspective and interested
I would like to be one of his supporters
I would also like to put forward a suggestion or idea because I use several drawing programs, including SketchBook
As the SketchBook program is characterized by the simplicity and strength of its tools at the same time
If you are interested, you can download the free version through the link below
https://www.autodesk.com/products/sketchbook/free-download.karikat
.................
It is worth noting that I suggested to interested developers to transfer the idea of SketchBook in the perspective tool to opentoonz, as it is very simple and intuitive, and at the same time its strength and accuracy
It is not like the one in clip studio paint, for example, "I used it too."
Clip Studio Paint is complex in perspective setting compared to SketchBook
Example from SketchBook:
![Autodesk SketchBook](https://user-images.githubusercontent.com/130619242/231916337-1d738c98-4301-410a-bccc-c9d275afaaf6.png)


Also suggest to replace "Ruler Tool" 
![1444-09-23 03_54_01-البريد](https://user-images.githubusercontent.com/130619242/231917535-1bbaab76-8609-4dfe-bc07-cbf66965bd09.png)
 to perspective icon, the ruler tool is also added with it.
Because the Ruler Tool is almost useless and I don't think it suffices.

I hope those interested will try SketchBook and check the perspective tool and how it works, as it is the simplest tool among all the programs I used
I think it is the closest to be used in opentoonz
* Finally, I hope that attention will be paid to adding the perspective tool in opentoonz in a simple way
Because our love for this program is its simplicity and power.
